src/ConnectedMode/QualityProfiles/OutOfDateQPFinder.cs

```diff
@@ -0,0 +1,115 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.Core;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Integration;
+using SonarQube.Client;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.ConnectedMode.QualityProfiles
+{
+    internal interface IOutOfDateQPFinder
+    {
+        /// <summary>
+        /// When in Connected Mode, returns the languages for which the Quality Profile
+        /// needs to be updated
+        /// </summary>
+        Task<IEnumerable<Language>> GetAsync(CancellationToken cancellationToken);
```

Comment:
Separate component to find the list of QPs that are out of date

Reply:
Currently, it returns the list of languages that need to be updated.
I was debating whether to return the server timestamp too (since we'll want to update the contents of the `binding.config` file with the updated timestamp). However, I decided to leave it for now and add it later if required.